stellargraph/data/unsupervised_sampler.py

```diff
@@ -56,23 +63,12 @@ def __init__(self, G, nodes=None, length=2, number_of_walks=1, seed=None):
             self.graph = G
 
         # Instantiate the walker class used to generate random walks in the graph
-        self.walker = UniformRandomWalk(G, seed=seed)
-
-        # This code will enable alternative walker classes
-        # TODO: Enable this code, but figure out how to pass required options to run
-        # if walker is not None:
-        #     if not isinstance(
-        #         walker, UniformRandomWalk
-        #     ):  # only work with Uniform Random Walker at the moment
-        #         raise TypeError(
-        #             "({}) Only Uniform Random Walks are possible".format(
-        #                 type(self).__name__
-        #             )
-        #         )
-        #     else:
-        #         self.walker = walker(G, seed=seed)
-        # else:
-        #         self.walker = UniformRandomWalk(G, seed=seed)
+        if walker is not None:
+            self.walker = walker
```

Comment:
Should we emit a warning or something if the other parameters are non-default (i.e. that they're being ignored)? I feel the `seed` one in particular is 'dangerous', since the user presumably intended to pass it to the walker when they constructed it (maybe without realising that's what they intended). Thoughts?

Reply:
That's a good point, I'll add warnings for when the other parameters are specified but ignored